### ISSUE TITLE
Adding info is crashing provider

### ISSUE THREAD

**Issue by FL3SH** [oboukili/terraform-provider-argocd#131]
```
    info {
      name  = "Repository"
      value = "foobar"
    }
```
```
Stack trace from the terraform-provider-argocd_v2.2.0 plugin:

panic: interface conversion: interface {} is map[string]interface {}, not map[string]string

goroutine 110 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationInfo(0xc000dec1e0, 0xc000726b40, 0x270e588, 0x4)
github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:372 +0x274
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSpec(0xc000d58c80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:39 +0x1a7
github.com/oboukili/terraform-provider-argocd/argocd.expandApplication(0xc000d58c80, 0xc0006a4440, 0xf, 0x0, 0x0, 0xc0006a4470, 0x6, 0x0, 0x0, 0x0, ...)
github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:21 +0x106
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDApplicationUpdate(0x2b105d8, 0xc000b76de0, 0xc000d58c80, 0x24d0060, 0xc000237ec0, 0xc0007c8150, 0x6fb0ea, 0xc000bc5060)
github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_application.go:246 +0x205
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000183a40, 0x2b10568, 0xc000c36000, 0xc000d58c80, 0x24d0060, 0xc000237ec0, 0x0, 0x0, 0x0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:364 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000183a40, 0x2b10568, 0xc000c36000, 0xc0003c8770, 0xc000bc5060, 0x24d0060, 0xc000237ec0, 0x0, 0x0, 0x0, ...)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0006d6288, 0x2b10568, 0xc000c36000, 0xc00101d770, 0xc000c36000, 0x25f46e0, 0xc001080800)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000bc40c0, 0x2b10610, 0xc000c36000, 0xc0003c85b0, 0xc000bc40c0, 0xc001080870, 0xc000bcfba0)
github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x25f46e0, 0xc000bc40c0, 0x2b10610, 0xc001080870, 0xc000934240, 0x0, 0x2b10610, 0xc001080870, 0xc000ff4d80, 0xc9b)
github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000299340, 0x2b29938, 0xc0008bf680, 0xc000fcfd40, 0xc00092cf00, 0x3e255e0, 0x0, 0x0, 0x0)
google.golang.org/grpc@v1.38.0/server.go:1286 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000299340, 0x2b29938, 0xc0008bf680, 0xc000fcfd40, 0x0)
google.golang.org/grpc@v1.38.0/server.go:1609 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0008181b0, 0xc000299340, 0x2b29938, 0xc0008bf680, 0xc000fcfd40)
google.golang.org/grpc@v1.38.0/server.go:934 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd

Error: The terraform-provider-argocd_v2.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by FL3SH** [comment]
Issue still occurs in 3.0.1

**Comment by onematchfox** [comment]
Fixed by #168
